### PR TITLE
Apply named "overrides" env/headers in config.yaml

### PR DESCRIPTION
**Pull request by kenigeer**
for Cloud Foundry support

### CODE REVIEW

File: lib/network.js [hunk 1]
```diff
@@ -384,6 +385,51 @@ function _setDefaults(config) {
...
+    applyOverrideValue(process.env[name], fn);
+  }
+
+
+  _.forEach(config.overrides, function(override) {
+
+    const flag = override.env && override.env.flag;
+    if (flag && process.env[flag]) {
+
+      applyEnvOverride(override.env.port, function(value) {
```
Comment (F1ERRO) [inline lib/network.js]: given that this is specified in the same config file as the `edgemicro` object, what does providing a port override in a separate top level object and then copying it over into `edgemicro` provide?

File: lib/network.js [hunk 2]
```diff
@@ -384,6 +385,51 @@ function _setDefaults(config) {
...
+      if (override.headers) {
+
+        applyOverrideValue(override.headers.retarget, function(value) {
+          console.info(`override ${override.name}: retarget_header = ${value}`)
+          config.edgemicro.retarget_header = value;
+        })
+
+      }
+
+      applyOverrideValue(override.log_to_console, function(value) {
```
Comment (F1ERRO) [inline lib/network.js]: this seems like something that should go into the top level `edgemicro.logging` object to begin with.

File: lib/network.js (hunk 1 above)
Comment (F1ERRO) [inline lib/network.js]: there is already a -p flag to override port

**Comment by kenigeer** [comment]
The idea is to provide an "atomic" *conditional* way to override some settings *if* edgemicro is being run in *different* kinds of containers. After configuring, you can run edgemicro locally, and can then push it and run it in the container without having to touch anything -- other than simply copying the ~/.edgemicro directory into the main directory in the case of Cloud Foundry. But you don't have to tinker with the files at all, since the override definition is in the default.yaml

Admittedly, only one implementation so far....

**Pull request closed without merging**